pkg/server/compression/compressor.go

```diff
@@ -0,0 +1,57 @@
+package compression
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+
+	"github.com/andybalholm/brotli"
+	"github.com/datadog/zstd"
+)
+
+type dummyCloser struct {
+	io.WriteCloser
+	w io.Writer
+}
+
+func NewDummyCloser(w io.Writer) io.WriteCloser {
+	return &dummyCloser{w: w}
+}
+
+func (d *dummyCloser) Write(p []byte) (n int, err error) {
+	return d.w.Write(p)
+}
```

Comment:
```suggestion
```
Not needed when embedding the io.Writer

Reply:
I removed dummyCloser entirely, and added a small branch in the server to check the non-compression part. It's smaller and probably less confusing.